backend_new/src/services/translation.service.ts

```diff
@@ -0,0 +1,167 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { LanguagesEnum, Prisma } from '@prisma/client';
+import { ListingGet } from '../dtos/listings/listing-get.dto';
+import { GoogleTranslateService } from './google-translate.service';
+import * as lodash from 'lodash';
+
+@Injectable()
+export class TranslationService {
+  constructor(
+    private prisma: PrismaService,
+    private readonly googleTranslateService: GoogleTranslateService,
+  ) {}
+
+  public async getTranslationByLanguageAndJurisdictionOrDefaultEn(
+    language: LanguagesEnum,
+    jurisdictionId: string | null,
+  ) {
+    try {
+      return await this.prisma.translations.findUniqueOrThrow({
+        where: {
+          jurisdictionId_language: { language, jurisdictionId },
+        },
+      });
+    } catch (e) {
+      if (
+        e instanceof Prisma.PrismaClientKnownRequestError &&
+        e.code === 'P2025'
+      ) {
+        console.warn(
+          `Fetching translations for ${language} failed, defaulting to english.`,
```

Comment:
can we add the jurisdiction id as well to this log or do you think it would be overkill?